src/peer_manager.rs

```diff
@@ -1217,11 +1219,14 @@ impl PeerManager {
 
         let _ = self.insert_peer(pub_id, Some(peer_id), PeerState::SearchingForTunnel);
 
-        let close_section = self.routing_table.other_close_names(pub_id.name()).unwrap_or_default();
+        let close_nodes = self.routing_table
```

Comment:
Wouldn't it be better to just return our entire section here (excluding ourself) and below filter the `peer_map` for those nodes which are in our section and are direct-connected to us and use `.take(NUM_TUNNEL_VIA_NODES)` there?